### DISCUSSION TITLE
Plugin and Presentation mode. Automatically on loading main patch

### DISCUSSION THREAD

**Discussion by pereshaped** [plugdata-team/plugdata#1788] [Ideas]
I've searched for this but could find an answer and wondered if it is possible to open a main patch in Plugin or Presentation mode?

Looked at the manual but it doesn't seem to mention it. Am I missing something obvious?, or is it not implemented yet? Somethin in the main environment inspector maybe? If not it would be a nice touch when deploying larger patches.

**Comment by timothyschoen** [comment]
Yes, this is possible, and no, you're not missing something obvious! This is a secret undocumented feature that really should be documented, sorry for that!

You can enter plugin mode by sending the "pluginmode" messages to "pd", like this:
```
#X obj 133 200 s pd;
#X msg 156 115 pluginmode;
#X connect 1 0 0 0;
```

The pluginmode message can also take an extra argument for the theme that must be loaded. This can be a theme name (like "fangs", "light", etc.) or a path to a custom '.plugdatatheme' file. This theme will only be applied temporarily, while you're in plugin mode. Hope that helps!

**Reply by pereshaped** [reply]
That is excellent news. That's perfect! Thanks!

**Reply by circusharmonic** [reply]
Hi, does this feature still work? I have copied your patch and I am getting the message the "pd: unknown message pluginmode" in the console

**Reply by dromer** [reply]
This should still work yes. Which version are you using?

**Comment by jyg** [comment]
That is great !!!!  Is there an opposite message to exit from  pluginmode ?

**Reply by dromer** [reply]
Not right now, but it would be neat if you could!

https://github.com/plugdata-team/plugdata/blob/develop/Source/PluginProcessor.cpp#L1497

**Reply by jyg** [reply]
Here is a PR :-)  https://github.com/plugdata-team/plugdata/pull/1858
However, wouldn't it be better if those messages were sent to a named canvas (like editmode message) instead of global pd receiver ?

**Reply by pereshaped** [reply]
It works better for me this way.
